providers/softlayer/src/main/java/org/jclouds/softlayer/compute/strategy/SoftLayerComputeServiceAdapter.java

```diff
@@ -272,7 +272,7 @@ public int compare(Hardware h1, Hardware h2) {
                     .compare(getBootableDeviceType(h1), getBootableDeviceType(h2));
             if (!volumes1.isEmpty() && !volumes2.isEmpty() && volumes1.size() == volumes2.size()) {
```

Comment:
@andreaturli Consider

h1: cores=1, ram=256, etc., volumes = [volume(type=FOO)]
h2: cores=1, ram=256, etc., volumes = []

These will compare to 0 currently.  I expect them to compare to either -1 or +1 since they are different.